internal/manifest/crypto_test.go

```diff
@@ -96,3 +96,29 @@ func newKey(t *testing.T, bits int) *rsa.PrivateKey {
 	t.Logf("generated key: %q", HexString(x509.MarshalPKCS1PrivateKey(key)))
 	return key
 }
+
+func TestSeedShareKeyParseMarshal(t *testing.T) {
+	keyData, err := NewSeedSharePrivateKey()
+	require.NoError(t, err)
+
+	key, err := ParseSeedshareOwnerPrivateKey(keyData)
+	require.NoError(t, err)
+
+	pubHexStr := MarshalSeedShareOwnerKey(&key.PublicKey)
+
+	_, err = ParseSeedShareOwnerKey(pubHexStr)
+	require.NoError(t, err)
+}
+
+func TestWorkloadOwnerKeyParseMarshal(t *testing.T) {
+	keyData, err := NewWorkloadOwnerKey()
+	require.NoError(t, err)
+
+	privateKey, err := ParseWorkloadOwnerPrivateKey(keyData)
+	require.NoError(t, err)
+
+	HashWorkloadOwnerKey(&privateKey.PublicKey)
+
+	_, err = ExtractWorkloadOwnerPublicKey(keyData)
```

Comment:
Could we add a test for the `PUBLIC KEY` case?

Reply:
Sure.